x/data/gas.go

```diff
@@ -0,0 +1,5 @@
+package data
+
+// TODO: Revisit this once we have proper gas fee framework.
+// Tracking issue https://github.com/cosmos/cosmos-sdk/discussions/9072
+const GasCostPerIteration = uint64(10)
```

Comment:
seams we are copying this value all over different places. Can we use it form a single place?
I know we are using a different modules, but maybe we can create some params module for that. (not necessary in this task). Something to discuss, not necessary do in this PR.

Reply:
I like the idea. Although if we want to set module specific values, we might want to keep them within each module, same way we do module specific errors. Currently we use `10` for all iterations but we might want to have some iterations charge more gas than others. I thought about making those changes here but figured I would start with making sure we are at least charging some gas for each iteration.